includes/class-debug.php

```diff
@@ -58,29 +66,101 @@ public static function log( $message, $type = 'string' ) {
 				error_log( 'AspireUpdate - Could not open or write to the file system. Check file system permissions to debug log directory.' );
 				// phpcs:enable
 			}
-			return;
+			return false;
 		}
+		return true;
+	}
 
-		$timestamp         = gmdate( 'Y-m-d H:i:s' );
-		$formatted_message = sprintf(
-			"[%s] [%s]: %s\n" . PHP_EOL,
-			$timestamp,
-			strtoupper( $type ),
-			self::format_message( $message )
-		);
-
-		$file_path = WP_CONTENT_DIR . '/' . self::$log_file;
+	/**
+	 * Get the content of the log file truncated upto N number of lines.
+	 *
+	 * @param integer  $limit Max no of lines to return. Defaults to a 1000 lines.
+	 *
+	 * @return string The File content truncate upto the number of lines set in the limit parameter.
+	 */
+	public static function read( $limit = 1000 ) {
+		$wp_filesystem = self::init_filesystem();
+		if ( self::verify_filesystem( $wp_filesystem ) ) {
+			$file_path = self::get_file_path();
+
+			if ( $wp_filesystem->exists( $file_path ) && $wp_filesystem->is_readable( $file_path ) ) {
+				$file_content = $wp_filesystem->get_contents_array( $file_path );
+				$content      = '';
+				$index        = 0;
+				foreach ( $file_content as $file_content_lines ) {
+					if ( ( $index < $limit ) ) {
+						$content .= $file_content_lines . PHP_EOL;
```

Comment:
The extra line in the "View Log" screen is caused by the `PHP_EOL` here.

With it:
![image](https://github.com/user-attachments/assets/ded8d73c-7514-4404-9a11-0cd76adc1983)


Without it:
![image](https://github.com/user-attachments/assets/05f23480-75b2-4b46-b0cb-b05fa0423777)

Reply:
It should be like image 1.  But I am getting another empty line there on Windows.

Reply:
Image 1 is how I want it.

Reply:
Can you try activating responses and requests too and see if you start to get two empty lines between each log entry.

Reply:
This is what I see in the file:
![image](https://github.com/user-attachments/assets/ddb1c60b-95b6-4d72-b9d6-476ea63f2861)

With the `PHP_EOL` still there, I get this in the viewer:
![image](https://github.com/user-attachments/assets/29a2f37a-5f9d-459f-b9b6-2c46cbc0b020)

After removing the `PHP_EOL`, I get this in the viewer:
![image](https://github.com/user-attachments/assets/dac80d2b-7fd8-4546-8220-d57af16217d4)

Reply:
This is happening because of an extra EOL being inserted by print_r.  We can leave it as it is for now.